app/controllers/vacancies_controller.rb

```diff
@@ -49,14 +49,14 @@ def set_headers
 
   def trigger_search_performed_event
     fail_safe do
-      vacancy_ids = @vacancies_search.vacancies.map(&:id)
+      vacancy_ids = @vacancies_search.vacancies.unscope(:order).pluck(:id)
       polygon_id = DfE::Analytics.anonymise(@vacancies_search.location_search.polygon.id) if @vacancies_search.location_search.polygon
 
       event_data = {
         search_criteria: form.to_hash,
         sort_by: form.sort.by,
         page: params[:page] || 1,
-        total_count: @vacancies_search.total_count,
+        total_count: vacancy_ids.size,
```

Comment:
This triggered the whole search again in DB to then count it.
With the changes on the `total_count` adding memoisation, this should be already memorised now, but having an array with all the result ids, we can pass the array size and ensure we won't be triggering a DB query.